src/password/scripts/directives.js

```diff
@@ -90,11 +89,13 @@
                         .then(function (response) {
                             scope.fields = response.data.actions.PUT;
                         }, function (response) {
-                            // If the response returns a 404 and we have a token.
-                            if (response.status === 404 && angular.isDefined(TOKEN)) {
-                                // There was a problem with the token.
+                            if (angular.isDefined(TOKEN)) {
                                 scope.tokenError = true;
                             }
+                            if (response.status !== 404 && angular.isDefined(response.data)) {
```

Comment:
Can we check what the API is returning here, and if it is definitely a 404? I think an error should really return 400